### PR TITLE
[ticket/16153] Enable Emojis and rich text in Topic title

### PR DESCRIPTION
**Pull request by 3D-I**
PHPBB3-16153

Checklist:

- [x] Correct branch: master for new features; 3.3.x & 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html), [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-16153

### CODE REVIEW

**Comment by 3D-I** [comment]
!set php

**Comment by 3D-I** [comment]
!set WIP

I will amend the test file.

**Comment by 3D-I** [comment]
!set Missing Tests

**Comment by 3D-I** [comment]
!unset Missing Tests

**Comment by 3D-I** [comment]
!unset WIP

File: phpBB/posting.php [hunk 1]
```diff
@@ -1181,11 +1181,23 @@
 		$error[] = $user->lang['EMPTY_SUBJECT'];
 	}
 
-	// Check for out-of-bounds characters that are currently
-	// not supported by utf8_bin in MySQL
+	/**
+	 * Replace Emojis and other 4bit UTF-8 chars not allowed by MySql to UCR / NCR.
```
Comment (marc1706) [inline phpBB/posting.php]:
```suggestion
	 * Replace Emojis and other 4bit UTF-8 chars not allowed by MySQL to UCR/NCR.
```

**Comment by SiteSplat** [comment]
This is just plain bad. Since when phpBB encourages users to spam? 
because this is what this PR blindly allows. Not going to even think about possible abuse of this.
At the very least there should be:

-forum based permission
-group based permission

and should be set to disabled by default.

If this is not possible then it should be reversed asap, please @VSEphpbb @marc1706

**Comment by DavidIQ** [comment]
How does this encourage users to spam?

**Comment by VSEphpbb** [comment]
I think emoji in titles is a welcomed addition.

**Comment by 3D-I** [comment]
> If the PR's author is not able to do so..

Do not make assumptions. And use the events to make it like what you want.

**Comment by SiteSplat** [comment]
Quite perplexing... I didn't expect I had to explain this further...  Don't get me wrong, but I gotta ask, Do you guys actually use forum in real life?

Do you think a board will look good with countless emojis like 💩 💩  💩  💩  in titles?
Do you know the repercussions in SEO?
Can you imagine on what this will look like in the topic list? something along this line:

**"hey low IQ this 💩  💩  💩 sucks lol lol crying face you need to waste time editing titles now?"**

**"🍽 No pasta today? ah that's why 👉 hungry"**

**"hey what's up cannot code a permission? oh that'"**

**"👋 this my 🥇 topic and pasta and 💩"**

**"The is 🔛 and the is 🍽 all the 🍩! wtf is this? 💩"**

There should even be a setting to limit the amount of emojis in titles, let alone permissions like I recommended above...

How, do you make "assumptions" this is a welcomed addition if you offer no way of controlling/choosing the outcome?

You are telling me that we can control:

-Can use topic/post icons
-Can use smilies

and many more,

But we won't offer control at all for emoji in titles? in fact we leave those enabled by default so we turn boards into kindergarten playgrounds?  

This is shortsighted to say the least.

**Comment by 3D-I** [comment]
No comments. I am out of this, no time to waste sorry.

**Comment by dark-1** [comment]
Emoji & rich text in titles is a welcomed & needed addition.

BUT if you want control over it ,
then start making Tickets & PRs ,

Everyone will be very happy & welcome you to add your valuable control setting for this...

Best regards 👍

**Comment by DavidIQ** [comment]
> Do you know the repercussions in SEO?

https://www.oberlo.com/blog/emoji-seo
https://cognitiveseo.com/blog/20067/emoji-seo-strategy/

Looks to me like emojis in titles would actually help your SEO. Sounds like a win-win!

**Comment by CrazyC0d3r** [comment]
While emoji support is a welcomed addition to phpBB, I agree with sitesplat that it should be permission based.

**Pull request merged**